ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -97,9 +97,17 @@ private void getDataStreamOutput(RaftClientRequest request, Set<RaftPeer> peers,
         throws IOException {
       for (RaftPeer peer : peers) {
         try {
-          outs.add((DataStreamOutputRpc) map.getProxy(peer.getId()).stream(request));
+          DataStreamClient client = map.getProxy(peer.getId());
+          outs.add((DataStreamOutputRpc) client.stream(request));
         } catch (IOException e) {
-          throw new IOException(map.getName() + ": Failed to getDataStreamOutput for " + peer, e);
+          map.handleException(peer.getId(), e, false);
+          throw new IOException(
+              map.getName() + ": Failed to getDataStreamOutput for " + peer, e);
+        } catch (NullPointerException e) {
```

Comment:
@guohao-rosicky , we should NEVER catch NullPointerException in production code since it indicates a bug.  Could you see what is the cause of it?

Reply:
@szetszwo I changed it to catch all exceptions and then trigger map.handleException

Reply:
I fixed it. Take a look. @szetszwo 
This problem I suspect is NettyServerStreamRpc#newClient when the peer is null.

Reply:
@guohao-rosicky , I mean we should not expect there is a NPE in runtime.  So, even catching Exception is incorrect.  We should fix the root cause of it.

Reply:
I agree with that. This PR will also solve other problems. I can open another JIRA and continue to follow the PROBLEMS of NPE.

What do you think? @szetszwo

Reply:
I changed this PR to handle when there is an exception in NettyServerStreamRpc#getDataStreamOutput

create new jira: NettyServerStreamRpc.Proxies#getProxy NPE

https://issues.apache.org/jira/browse/RATIS-1531

Reply:
@guohao-rosicky , this is not about the title of the JIRA.  Please see how could we fix the NPE.  We really should not catch Exception in NettyServerStreamRpc.  It hides a bug instead of fixing it.

Reply:
I agree, I will submit another PR later to solve the NPE problem, this PR can be changed to catch IOException

Reply:
NPE is a problem when initializing Client connections. I will add assertions or retry in another PR.
@szetszwo